phpcs.xml.dist

```diff
@@ -10,6 +10,11 @@
 	<rule ref="WordPress.WP.I18n"/>
 	<config name="text_domain" value="gutenberg,default"/>
 
+	<rule ref="VariableAnalysis.CodeAnalysis.VariableAnalysis">
+		<properties>
+			<property name="allowUnusedFunctionParameters" value="true"/>
```

Comment:
What about using `allowUnusedParametersBeforeUsed` instead?

Reply:
Hi @swissspidy, the rule was causing problems when we need in a class to overload a function defined in the base class without necessarily using all the parameters.
But on second thought these cases are rare and we should not optimize for them, so I followed your suggestion and disabled the rule on the specific cases where it was problematic.